pkg/cloud/openstack/clients/machineservice.go

```diff
@@ -342,7 +352,34 @@ func GetSecurityGroups(is *InstanceService, sg_param []openstackconfigv1.Securit
 	return sgIDs, nil
 }
 
-func (is *InstanceService) InstanceCreate(clusterName string, name string, config *openstackconfigv1.OpenstackProviderSpec, cmd string, keyName string) (instance *Instance, err error) {
+// Helper function for getting image ID from name
+func getImageID(is *InstanceService, imageName string) (string, error) {
+	opts := images.ListOpts{
+		Name: imageName,
+	}
+
+	pages, err := images.List(is.imagesClient, opts).AllPages()
+	if err != nil {
+		return "", err
+	}
+
+	allImages, err := images.ExtractImages(pages)
+	if err != nil {
+		return "", err
+	}
+
+	switch len(allImages) {
+	case 0:
+		return "", fmt.Errorf("no image with the name %s could be found", imageName)
+	case 1:
+		return allImages[0].ID, nil
+	default:
+		return "", fmt.Errorf("too many images with the name, %s, were found", imageName)
+	}
+}
+
+// InstanceCreate creates a compute instance
+func (is *InstanceService) InstanceCreate(clusterName string, name string, config *openstackconfigv1.OpenstackProviderSpec, cmd string, keyName string, cluster *openstackconfigv1.OpenstackClusterProviderSpec) (instance *Instance, err error) {
```

Comment:
I would like to change to order or parameters. And rename param `cluster` to `clusterSpec`.
`clusterName string, name string, clusterSpec *openstackconfigv1.OpenstackClusterProviderSpec, config *openstackconfigv1.OpenstackProviderSpec,...`, just because of convenience. The `machine.Actuator` also references cluster before machine.